app/src/main/java/com/babylon/wallet/android/presentation/dapp/unauthorized/verifyentities/VerifyEntitiesViewModel.kt

```diff
@@ -0,0 +1,176 @@
+package com.babylon.wallet.android.presentation.dapp.unauthorized.verifyentities
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import com.babylon.wallet.android.data.dapp.IncomingRequestRepository
+import com.babylon.wallet.android.di.coroutines.DefaultDispatcher
+import com.babylon.wallet.android.domain.model.messages.WalletUnauthorizedRequest
+import com.babylon.wallet.android.domain.model.signing.SignPurpose
+import com.babylon.wallet.android.domain.model.signing.SignRequest
+import com.babylon.wallet.android.presentation.accessfactorsources.AccessFactorSourcesInput
+import com.babylon.wallet.android.presentation.accessfactorsources.AccessFactorSourcesProxy
+import com.babylon.wallet.android.presentation.common.OneOffEvent
+import com.babylon.wallet.android.presentation.common.OneOffEventHandler
+import com.babylon.wallet.android.presentation.common.OneOffEventHandlerImpl
+import com.babylon.wallet.android.presentation.common.StateViewModel
+import com.babylon.wallet.android.presentation.common.UiState
+import com.radixdlt.sargon.SignatureWithPublicKey
+import com.radixdlt.sargon.extensions.ProfileEntity
+import com.radixdlt.sargon.extensions.asProfileEntity
+import com.radixdlt.sargon.extensions.hex
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import rdx.works.core.sargon.activeAccountsOnCurrentNetwork
+import rdx.works.core.sargon.activePersonaOnCurrentNetwork
+import rdx.works.profile.domain.GetProfileUseCase
+import javax.inject.Inject
+
+@HiltViewModel
+class VerifyEntitiesViewModel @Inject constructor(
```

Comment:
The `VerifyPersonaScreen` and `VerifyAccountsScreen` internally share the `VerifyEntitiesContent` composable since those screens are quite similar.

However the two screens use the same `VerifyEntitiesViewModel`. **Not as a shared viewmodel**, but one instance per screen.